src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,47 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
+  checkHealth() {
+    if (this.health <= 0) {
+      const index = Animal.alive.indexOf(this);
+
+      if (index > -1) {
+        Animal.alive.splice(index, 1);
+      }
```

Comment:
[CODE PERFORMANCE] Instead of using `indexOf` and `splice` to remove dead animals from the `Animal.alive` array, which can be inefficient due to nesting loops, consider using the `filter` method to create a new array that only includes living animals. This approach avoids the nested loop and can be more performant.